pkg/controller/projectreference/projectreference_adapter.go

```diff
@@ -318,23 +309,21 @@ func (r *ReferenceAdapter) createProject(parentFolderID string) error {
 		case "DELETE_REQUESTED":
 			return operrors.ErrInactiveProject
 		default:
-			r.logger.Error(operrors.ErrUnexpectedLifecycleState, "Unexpected LifecycleState", project.LifecycleState)
-			return operrors.ErrUnexpectedLifecycleState
+			return operrors.Wrap(operrors.ErrUnexpectedLifecycleState, fmt.Sprintf("unexpected lifecycleState for %s", project.LifecycleState))
 		}
 	}
 
 	r.logger.V(int(logtypes.ProjectReference)).Info("Creating Project")
 	// If we cannot create the project clear the projectID from spec so we can try again with another unique key
 	_, err = r.gcpClient.CreateProject(parentFolderID)
 	if err != nil {
-		r.logger.Error(err, "could not create project", "Parent Folder ID", parentFolderID, "Requested Project ID", r.ProjectReference.Spec.GCPProjectID)
 		r.logger.V(int(logtypes.ProjectReference)).Info("Clearing gcpProjectID from ProjectReferenceSpec")
 		//Todo() We need to requeue here ot it will continue to the next step.
 		err = r.clearProjectID()
 		if err != nil {
 			return err
 		}
-		return err
+		return operrors.Wrap(err, fmt.Sprintf("could not create project. Parent Folder ID: %s, Requested Project ID: %s", parentFolderID, r.ProjectReference.Spec.GCPProjectID))
```

Comment:
Can you explain more why `err` should be nil here? 
Shouldn't we report project creation errors if it isn't being created properly?

```go
	_, err = r.gcpClient.CreateProject(parentFolderID)
	if err != nil {
		r.logger.V(int(logtypes.ProjectReference)).Info("Clearing gcpProjectID from ProjectReferenceSpec")
		//Todo() We need to requeue here ot it will continue to the next step.
		err = r.clearProjectID()
		if err != nil {
			return err
		}
		return operrors.Wrap(err, fmt.Sprintf("could not create project. Parent Folder ID: %s, Requested Project ID: %s", parentFolderID, r.ProjectReference.Spec.GCPProjectID))
	}
```

Reply:
mm we may need to rename the first `err` :D

Reply:
@georgettica can you review again? I've removed the ambiguity by renaming first error

Reply:
@boranx still, let me try to explain with this code
```
func foo() error {
  err := bar()
  if err !=nil {
    return err
  } 
  return err // In this line err will be nil
}
```
as you can see, because I am returning inside the `if` statement the result in the second return must be nil

Reply:
https://github.com/openshift/gcp-project-operator/pull/77/files#diff-7dfee1ca5a86e286b7a28877f7cf8f80R318

Reply:
thanks, it didn't show up on my side until now. looks good

Reply:
I've already fixed there

```go
	_, creationFailed := r.gcpClient.CreateProject(parentFolderID)
	if creationFailed != nil {
		r.logger.V(int(logtypes.ProjectReference)).Info("Clearing gcpProjectID from ProjectReferenceSpec")
		//Todo() We need to requeue here ot it will continue to the next step.
		if err = r.clearProjectID(); err != nil {
			return operrors.Wrap(creationFailed, fmt.Sprintf("could not clear project ID: %v", err))
		}

		return operrors.Wrap(creationFailed, fmt.Sprintf("could not create project. Parent Folder ID: %s, Requested Project ID: %s", parentFolderID, r.ProjectReference.Spec.GCPProjectID))
	}
```